docs/extend/extend-apiml/api-mediation-oidc-authentication.md

```diff
@@ -0,0 +1,98 @@
+
+- Users have been assigned to the Client Application.
+
+  To access mainframe resources, users with a distributed authentication must either be directly assigned by the OIDC provider to the client application, or must be part of group which is allowed to work with the client application.     
+
+### ESM configuration 
+The user identity mapping is defined as a distributed user identity filter, which is maintained by the System Authorization Facility (SAF) / External Security Manager (ESM).
+A distributed identity consists of two parts: a distributed identity name, and a trusted registry which governs that identity. 
+API ML provides a Zowe CLI plugin to help administrators to generate a JCL for creating the mapping filter specific for the ESM installed on the target mainframe system. 
+
+  See the [Identity Federation cli plugin]( )  <!--Add link --> documentation for details about how to use the plugin tool to set up the mapping in the ESM of your z/OS system.
+
+Alternatively, administrators can use the installed ESM functionality to create, delete, list, and query a distributed identity filter/s:
+ - For RACF consult [RACMAP command](https://www.ibm.com/docs/en/zos/2.3.0?topic=rcs-racmap-create-delete-list-query-distributed-identity-filter).
+ - For CA Top Secret use the [IDMAP Keyword - Implement z/OS Identity Propagation Mapping](https://techdocs.broadcom.com/us/en/ca-mainframe-software/security/ca-top-secret-for-z-os/16-0/administrating/issuing-commands-to-communicate-administrative-requirements/keywords/idmap-keyword-implement-z-os-identity-propagation-mapping.html).
+ - For CA ACF2 use [IDMAP User Profile Data Records](https://techdocs.broadcom.com/us/en/ca-mainframe-software/security/ca-acf2-for-z-os/16-0/administrating/administer-records/user-profile-records/idmap-user-profile-records.html).
+
+## API ML configuration
+Use the following procedure to enable the feature to use an OIDC Access Token as the method of authentication for the API Mediation Layer Gateway.
+
+ In the zowe.yaml file, configure the following properties:
+          
+   * **components.gateway.apiml.security.oidc.enabled**  
+   Specifies the global feature toggle. Set the value to true to enable OIDC authentication functionality.
+
+   * **components.gateway.apiml.security.oidc.clientId**  
+   Specifies the value of the client identification (client_id) assigned by the OIDC provider to the API ML Gateway.
+   
+   * **components.gateway.apiml.security.oidc.clientSecret**   
+   Specifies the client secret assigned by the OIDC provider to the API ML Gateway. It is used in combination with the client_id in Access Token validation requests at the /introspection endpoint of the OIDC provider.
+   
+   * **components.gateway.apiml.security.oidc.registry**  
+   Specifies the SAF registry used to group the identities recognized as having a OIDC identity mapping.        
+
```

Comment:
@PetrWeinfurt: Do we need a property for the ZWESVUSR name to be assigned for the identity mapper, similarly to the x.509 mapper?

Reply:
I don't understand this whole **Note** 
 - the security configuration for the identityMapperUser has been already discussed in section above
 - If the Zowe runtime user name has been changed does not really matter ... the same name will be used also for this property too. The identityMapperUser property exist for situations where the sysprog wants to use other user for the mapping queries than the Zowe runtime user.